templates/cargo/github/test.yml

```diff
@@ -330,23 +343,143 @@ jobs:
 
     - name: Run grcov
       run: |
-        grcov . --binary-path ./target/debug/ -t coveralls -s . --token YOUR_COVERALLS_TOKEN -o coveralls.json
+        grcov . --binary-path ./target/debug/ --ignore "**/.cargo/**" --ignore "**/rustc/**" --ignore "./target/**" -t coveralls -s . --token YOUR_COVERALLS_TOKEN > coveralls.json
 
     - name: Run weighted-code-coverage
+      id: wcc-run
       run: |
         mkdir $HOME/wcc-output
-        weighted-code-coverage -p src/ -j coveralls.json -c cyclomatic --json $HOME/wcc-output/out.json
+        echo "html=$HOME/wcc-output" >> $GITHUB_OUTPUT
+        echo "json=$HOME/wcc-output/wcc.json" >> $GITHUB_OUTPUT
+        weighted-code-coverage --project-path ./ --grcov-format coveralls --grcov-path ./coveralls.json -m functions --json $HOME/wcc-output/wcc.json --html $HOME/wcc-output/
 
-    - name: Upload weighted-code-coverage data
-      uses: actions/upload-artifact@v3
+    - name: Setup GitHub Pages
+      uses: actions/configure-pages@v5.0.0
+      with:
+        enablement: true
+
+    - name: Upload Wcc html output
+      uses: actions/upload-pages-artifact@v3.0.1
       with:
-        name: weighted-code-coverage-ubuntu
-        path: ~/wcc-output/out.json
+        path: {{ '${{ steps.wcc-run.outputs.html }}' }}
+
+    - name: Deploy to GitHub Pages
+      id: deployment
+      uses: actions/deploy-pages@v4.0.5
+
+    - name: Extract metric values
+      id: metrics
+      env:
+        JSON: {{ '${{ steps.wcc-run.outputs.json }}' }}
+      run: |
+        content=`cat ${JSON}`
+        echo "json=$content" >> $GITHUB_OUTPUT
+
+    - name: Extract branch name
+      id: extract-branch
+      if:  github.event_name != {{ '\'pull_request\'' }}
```

Comment:
Is this necessary?  Can we just write `if:  github.event_name != 'pull_request'`? Or use another way to tell that we are in a PR?

Reply:
It works with `if: github.event_name != 'pull_request'`.